### ISSUE TITLE
Rework error handling for synchronous file system module API

### ISSUE THREAD

**Issue by manoldonev** [NativeScript/NativeScript#6260]
Currently the error handling for the [synchronous file system module API](https://docs.nativescript.org/ns-framework-modules/file-system#read) is performed via optional error callbacks that are passed as last parameter to the respective "sync" method. This approach is rather unconventional and unintuitive:

``` ts
file.writeTextSync("some text", (err) => console.log(err));
```

Rework the error handling logic for the "sync" methods to be similar to the [nodejs approach](https://nodejs.org/api/fs.html) with try...catch blocks:

```
try {
    file.writeTextSync("some text");
} catch (err) {
    console.log(err);
}
```

**Comment by vakrilov** [comment]
Totally agree  the API with `try/catch` is much better.

Maybe we can implement a back-compatible approach. For example, if an error occurs
- If there is `errorCallback` - just call it and leave the user to handle it (the old way)
- If there is not -> throw the error (the new way)

**Comment by manoldonev** [comment]
Yup, we can do that but I would still remove the "old" way from our documentation / examples.

**Comment by NathanaelA** [comment]
Please support the old way; there are several plugins in the ecosystem that use the FS module for things; and I know I don't really want to make the plugins incompatible with people still using NS 4...
